src/uu/pmap/src/maps_format_parser.rs

```diff
@@ -47,13 +99,13 @@ fn parse_address(memory_range: &str) -> (String, u64) {
     (format!("{start:0>16}"), size_in_kb)
 }
 
-// Turns a 4-char perms string from /proc/<PID>/maps into a 5-char perms string. The first three
-// chars are left untouched.
-fn parse_perms(perms: &str) -> String {
-    let perms = perms.replace("p", "-");
-
-    // the fifth char seems to be always '-' in the original pmap
-    format!("{perms}-")
+// Pads the device info from /proc/<PID>/maps with zeros and turns AB:CD into 0AB:000CD.
+fn parse_device(device: &str) -> String {
+    if let Some((major, minor)) = device.split_once(':') {
+        format!("{major:0>3}:{minor:0>5}")
+    } else {
+        unreachable!("device must contain a ':'")
```

Comment:
unreachable will panic

maybe just something like
        Err(Error::new(ErrorKind::InvalidData, "Invalid device format"))

Reply:
Hm. I wrote `maps_format_parser.rs` with the assumption that the value passed to `parse_map_line()` has the correct format ;-)  It's the reason it returns a `MapLine` and not a `Result` and there are also some `expect`s that will panic. So should I write it more defensively?